File: app/src/main/java/com/junkiedan/junkietuner/data/dao/TuningDao.java
package com.junkiedan.junkietuner.data.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.junkiedan.junkietuner.data.entities.Tuning;

import java.util.List;

@Dao
public interface TuningDao {
    @Query("SELECT * FROM Tuning")
    LiveData<List<Tuning>> getAllTunings();

    @Query("SELECT * FROM Tuning WHERE id=:id")
    LiveData<Tuning> getTuningById(int id);

    @Update
    void update(Tuning tuning);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(Tuning... tunings);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertOne(Tuning tuning);

    @Delete
    void delete(Tuning tuning);

    @Query("DELETE FROM Tuning")
    void deleteAll();
}

File: app/src/main/java/dev/thestbar/tunify/data/dao/TuningDao.kt
package dev.thestbar.tunify.data.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import dev.thestbar.tunify.data.entities.Tuning
import kotlinx.coroutines.flow.Flow

@Dao
interface TuningDao {

    @Query("SELECT * FROM Tuning")
    fun getAllTunings(): Flow<List<Tuning>>

    @Query("SELECT * FROM Tuning WHERE id = :id")
    fun getTuningById(id: Int): Flow<Tuning?>

    @Update
    suspend fun update(tuning: Tuning)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAll(vararg tunings: Tuning)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertOne(tuning: Tuning)

    @Delete
    suspend fun delete(tuning: Tuning)

    @Query("DELETE FROM Tuning")
    suspend fun deleteAll()
}
